synapse/storage/database.py

```diff
@@ -394,8 +394,8 @@ def execute_values(
             sql,
         )
 
-    def execute(self, sql: str, *args: Any) -> None:
-        self._do_execute(self.txn.execute, sql, *args)
+    def execute(self, sql: str, parameters: _Parameters = ()) -> None:
+        self._do_execute(self.txn.execute, sql, parameters)
 
     def executemany(self, sql: str, *args: Any) -> None:
```

Comment:
executemany is also presumably vulnerable to this?

Reply:
I left some passive aggressive comments for our future selves.

Reply:
(I have in mind the bugs we've seen where we retry a transaction with a different set of parameters because we've exhausted an iterable. But I think that's our own retry code---idk if the db drivers worry about this at all?)

Reply:
Yeah, probably. Should we just do it while we're here?

Reply:
I tried this but found that it caused an explosion of errors. (I meant to stick those errors in a comment but I seem to have forgotten to do this.)

Reply:
Instead of a comment can we file an issue?

Reply:
-> https://github.com/matrix-org/synapse/issues/15439